packages/mobile/src/account/Settings.tsx

```diff
@@ -258,9 +263,36 @@ export class Account extends React.Component<Props, State> {
     ValoraAnalytics.track(SettingsEvents.tos_view)
   }
 
+  onRemoveAccountPress = () => {
+    this.setState({ showAccountKeyModal: true })
+  }
+
+  hideRemoveAccountModal = () => {
+    this.setState({ showAccountKeyModal: false })
+  }
+
+  onPressContinueWithAccountRemoval = () => {
+    this.setState({ showAccountKeyModal: false })
+    this.props.navigation.navigate(Screens.BackupPhrase, { fromSettings: true })
+  }
+
+  hideConfirmRemovalModal = () => {
+    this.props.navigation.setParams({ promptConfirmRemovalModal: false })
+  }
+
+  confirmAccountRemoval = async () => {
+    await this.props.clearStoredAccount(this.props.account || '')
```

Comment:
I don't think this does what you expect it do. This just synchronously dispatches the action to the store which, if you've set up the watcher - will have Redux Saga trigger the saga generator. Nothing happens if you await it.